src/pages/auth/components/GoogleLoginButton.tsx

```diff
@@ -0,0 +1,39 @@
+import { Box } from '@mui/material';
+import { GoogleLogin } from '@react-oauth/google';
+import useGoogleSignInMutation from '@src/api/mutations/auth/useGoogleSignInMutation';
+import i18n from '@src/i18n/i18n';
+import { ErrorTypes } from '@src/types/i18n.types';
+import { useCallback } from 'react';
+import toast from 'react-hot-toast';
+
+const GoogleLoginButton = () => {
+  const { mutate: googleLogin } = useGoogleSignInMutation();
+
+  const handleGoogleLogin = useCallback(
+    (token: string) => {
+      if (token) {
+        googleLogin(token);
+      }
+    },
+    [googleLogin]
+  );
+
+  return (
+    <Box display="flex" justifyContent="center" margin={4}>
+      <GoogleLogin
+        onSuccess={({ credential }) => {
+          if (!credential) {
```

Comment:
you can move this line to `handleGoogleLogin`

Reply:
done